lib/ops/opsservice/site.go

```diff
@@ -644,6 +647,53 @@ func (s site) gid() string {
 	return defaults.ServiceUserID
 }
 
+func (s *site) getClusterConfiguration() (*clusterconfig.Resource, error) {
+	client, err := s.service.GetKubeClient()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	configmap, err := client.CoreV1().ConfigMaps(defaults.KubeSystemNamespace).
+		Get(constants.ClusterConfigurationMap, metav1.GetOptions{})
+	err = rigging.ConvertError(err)
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	}
+	var spec string
+	if configmap != nil {
+		spec = configmap.Data["spec"]
+	}
+	var config *clusterconfig.Resource
+	if spec != "" {
+		config, err = clusterconfig.Unmarshal([]byte(spec))
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	} else {
+		config = clusterconfig.NewEmpty()
+	}
+	if err := s.setClusterConfigDefaults(config); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return config, nil
+}
+
+func (s *site) setClusterConfigDefaults(config *clusterconfig.Resource) error {
+	if config.Spec.Global.CloudProvider == "" {
+		config.Spec.Global.CloudProvider = s.provider
+	}
+	installOp, _, err := ops.GetInstallOperation(s.key, s.service)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if config.Spec.Global.PodCIDR == "" {
+		config.Spec.Global.PodCIDR = installOp.InstallExpand.Vars.OnPrem.PodCIDR
```

Comment:
Nit: I'd check for `InstallExpand` being not nil, even though this is an install operation.